test.sh

```diff
@@ -12,20 +13,35 @@ This script creates virtual network adapter which can be used for bridging
 Usage:
 $0 [options]
 Options:
-    start   This option will create virtual adapter
-    stop    This option will destory virtual adapter
+    -h, --help  Display this help menu
+    start       This option will create virtual adapter
+    stop        This option will destory virtual adapter
     
 _EOF
 }
 
+if [[ "$1" == "-h" ]] || [[ "$1" == "--help" ]] ; then
+    help
+    exit 0
+fi
+
+if [[ "$DISTRO" == "Ubuntu" ]] || [[ "$DISTRO" == "LinuxMint" ]] ; then
+    use_sudo="sudo"
+fi
+
+if [[ "$use_sudo" == "" ]] && [[ "$UID" != 0 ]] ; then
+    echo "error: execute script as root user"
+    exit 1
+fi
```

Comment:
You can check for the command `sudo` like this:
```bash
if [[ -z $(which sudo) ]]; then
    echo "Please install sudo first. For example see this tutorial:"
    echo "https://www.vultr.com/docs/how-to-use-sudo-on-debian-centos-and-freebsd"
    exit 1
fi
```

And that's all.

Reply:
Very nice!